src/manifest.rs

```diff
@@ -1,21 +1,97 @@
-pub fn get_package_name(manifest_path: &std::path::Path) -> anyhow::Result<String> {
-    let manifest = std::fs::read_to_string(manifest_path).map_err(|e| {
-        anyhow::format_err!("Failed when reading {}: {}", manifest_path.display(), e)
-    })?;
-    let manifest: toml_edit::Document = manifest
-        .parse()
-        .map_err(|e| anyhow::format_err!("Failed to parse {}: {}", manifest_path.display(), e))?;
-    let package = manifest.get("package").ok_or_else(|| {
+#[derive(Debug, Clone)]
+pub(crate) struct Manifest<'a> {
+    pub(crate) path: &'a std::path::Path,
+    pub(crate) doc: toml_edit::Document,
+}
+
+impl<'a> Manifest<'a> {
+    pub(crate) fn parse(path: &'a std::path::Path) -> anyhow::Result<Self> {
+        let manifest_text = std::fs::read_to_string(path)
+            .map_err(|e| anyhow::format_err!("Failed when reading {}: {}", path.display(), e))?;
+        let doc = manifest_text
+            .parse()
+            .map_err(|e| anyhow::format_err!("Failed to parse {}: {}", path.display(), e))?;
```

Comment:
Shouldn't most of this be done for us with `cargo_metadata`?

Reply:
These lines are just moved and extracted into their own method, see above.

Reply:
After looking into it a bit, it seems that `metadata.workspace_packages()` can give us a `Vec<&Package>` and the `Package` type has the info on targets we need.

However, I'm not sure how to pick out the correct `&Package` from that vector based on the `pkg_spec: Option<&str>` function parameter. In general I'm also not familiar enough with cargo's data model (e.g. specs work) so I'm not that comfortable or effective with this code.

If you're able to take a look at it (and you feel it's worth looking into), that would be awesome.